pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableRestletResource.java

```diff
@@ -558,4 +558,54 @@ public RebalanceResult rebalance(
       throw new ControllerApplicationException(LOGGER, e.getMessage(), Response.Status.NOT_FOUND);
     }
   }
+
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("/tables/{tableName}/state")
+  @ApiOperation(value = "Get current table state", notes = "Get current table state")
+  public String getTableState(
+      @ApiParam(value = "Name of the table to get its state", required = true) @PathParam("tableName") String tableName,
+      @ApiParam(value = "realtime|offline") @QueryParam("type") String tableTypeStr
```

Comment:
Let's just make type as required, and keep return type binary, instead of introducing a json blurb here.

Reply:
Just 0 or 1?

Reply:
Can still be enabled/disabled or true/false. But just single string with only 2 possible options

Reply:
Done, updated the return to disabled/enabled.
Wouldn't be better if we create an enum TableState with these values [enabled/disabled]?

Reply:
yes certainly. We already have one called StateType I believe. Wonder if we can reuse

Reply:
Yeah, I saw it, but I guess we should leave it as just these simple strings [disabled/enabled].